SonarQube.Client.Tests/Helpers/SecondaryIssueHashUpdaterTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarQube.Client.Models;
+
+namespace SonarQube.Client.Helpers.Tests
+{
+    [TestClass]
+    public class SecondaryIssueHashUpdaterTests
+    {
+        private const string PrimaryIssueHash = "the primary issue hash should not be affected";
+        private readonly IChecksumCalculator checksumCalculator = new ChecksumCalculator();
+
+        [TestMethod]
+        public async Task Populate_NoIssues_NoOp()
+        {
+            var serviceMock = new Mock<ISonarQubeService>();
+
+            var testSubject = new SecondaryIssueHashUpdater();
+            await testSubject.UpdateHashesAsync(Array.Empty<SonarQubeIssue>(), serviceMock.Object, CancellationToken.None);
+
+            serviceMock.Invocations.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public async Task Populate_NoSecondaryLocations_NoOp()
+        {
+            var serviceMock = new Mock<ISonarQubeService>();
+            var issues = new[]
+            {
+                CreateIssue("project1:key1", AddFlow()),
```

Comment:
I don't think we can have a flow with no locations in reality.
Perhaps one of the issues can be with empty flows, to simulate the real conditions.